sensu/plugins/metrics-keystone.py

```diff
@@ -0,0 +1,75 @@
+#/usr/bin/env python
+# -*- coding: utf-8 -*-
+from __future__ import print_function
+
+import time
+from collections import namedtuple
+
+import shade
+import os_client_config
+from keystoneclient import client
+from keystoneauth1.session import Session
+
+OK = 0
+WARNING = 1
+CRITICAL = 2
+UNKNOWN = 3
+
+
+class CloudMetrics(object):
+    def __init__(self):
+        self.cloud = shade.openstack_cloud()
+        self.keystone = None
+        self._setup()
+
+    def _setup(self):
+        config = os_client_config.OpenStackConfig()
+        cloud_config = config.get_one_cloud()
+        session = Session(cloud_config.get_auth())
+        self.keystone = client.Client(session=session)
```

Comment:
With shade you should have access to `cloud.keystone_client` and you don't have to set up your own keystone client object.